docs/Allocene/script/games.js

```diff
@@ -0,0 +1,189 @@
+    return params[selectedGenreNow] || params.allgames;
+}
+
+function buildDynamicUrl(params) {
+    const dynamicUrl = genreFilter(params);
+    const platform = document.querySelector('[data-type="platform-radio"]').checked
+        ? params.platform
+        : '';
+    const online = document.querySelector('[data-type="online-radio"]').checked
+        ? params.online
+        : '';
+    const newGames = document.querySelector('[data-type="check-new"]').checked
+        ? params.newGames
+        : '';
+    const oldGames = document.querySelector('[data-type="check-old"]').checked
+        ? params.oldGames
+        : '';
+
+    return `${params.genre}${dynamicUrl}${platform}${online}${newGames}${oldGames}`;
+}
+
+async function applyFiltersAndRenderCards() {
+    const dynamicUrl = buildDynamicUrl(queryParams);
+    await renderCards(ulContainer, dynamicUrl);
+}
+
+const selectElement = document.getElementById('genre');
+selectElement.addEventListener('change', async () => {
+    await applyFiltersAndRenderCards();
+});
+
+genreFilter(queryParams);
+
+function radioFilter(params) {
+    radioPlatform.addEventListener('change', async () => {
+        params.dynamicUrl = radioPlatform.checked ? params.platform : params.allgames;
+        await applyFiltersAndRenderCards();
+    });
+
+    radioOnline.addEventListener('change', async () => {
+        params.dynamicUrl = radioOnline.checked ? params.online : params.allgames;
+        await applyFiltersAndRenderCards();
+    });
+}
+
+function checkboxFilter(params) {
+    let dynamicUrl = params.allgames;
+
+    checkNew.addEventListener('change', async () => {
+        dynamicUrl = checkNew.checked ? params.newGames : params.allgames;
+        await applyFiltersAndRenderCards();
+    });
+
+    checkOld.addEventListener('change', async () => {
+        dynamicUrl = checkOld.checked ? params.oldGames : params.allgames;
+        await applyFiltersAndRenderCards();
+    });
+
+    renderCards(ulContainer, dynamicUrl);
+}
+
+function searchFilter() {
+    searchField.addEventListener('input', () => {
+        const searchTerm = searchField.value.toLowerCase();
+        const cards = document.querySelectorAll('.game__cod2');
+
+        let anyCardFound = false;
+
+        cards.forEach((card) => {
+            const title = card.querySelector('[data-type="title"]').textContent.toLowerCase();
+            const description = card.querySelector('[data-type="description"]').textContent.toLowerCase();
+
+            if (title.includes(searchTerm) || description.includes(searchTerm)) {
+                card.style.display = 'block';
+                anyCardFound = true;
+            } else {
+                card.style.display = 'none';
+            }
+        });
+        noFound.innerText = anyCardFound ? '' : 'No results found!';
+    });
+}
+
+async function callFunctionsAndRenderCards(params) {
+    await applyFiltersAndRenderCards(params);
+    genreFilter(params);
```

Comment:
```suggestion
    addFiltersHandlers() // in it add all handlers
```

Reply:
I did not understand the point of the edit